modules/products/bazel/src/main/kotlin/com/metalbear/mirrord/products/bazel/BazelExecutionListener.kt

```diff
@@ -0,0 +1,154 @@
+                    executionInfo.patchedPath?.let {
+                        MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: patchedPath is not null: $it, meaning original was SIP")
+                        state.blazeBinaryState.blazeBinary = it
+                    }
+                }
+
+                savedEnvs[executorId] = SavedConfigData(originalEnv, originalBazelPath)
+            }
+        } catch (e: Throwable) {
+            MirrordLogger.logger.debug("[${this.javaClass.name}] processStartScheduled: exception catched: ", e)
+            // Error notifications were already fired.
+            // We can't abort the execution here, so we let the app run without mirrord.
+            service.notifier.notifySimple(
+                "Cannot abort run due to platform limitations, running without mirrord",
+                NotificationType.WARNING
+            )
+        }
+
+        super.processStartScheduled(executorId, env)
+    }
+
+    /**
+     * Restores original configuration after Bazel run has ended.
+     */
+    private fun restoreConfig(executorId: String, configState: BlazeCommandRunConfigurationCommonState) {
+        MirrordLogger.logger.debug("[${this.javaClass.name}] restoreEnv: $executorId $configState")
+
+        val saved = savedEnvs.remove(executorId) ?: run {
+            MirrordLogger.logger.debug("[${this.javaClass.name}] restoreConfig: no saved env found")
+            return
+        }
+
+        MirrordLogger.logger.debug("[${this.javaClass.name}] restoreConfig: found ${saved.envVars.size} saved original variables")
+        configState.userEnvVarsState.setEnvVars(saved.envVars)
+
+        if (SystemInfo.isMac) {
+            MirrordLogger.logger.debug("[${this.javaClass.name}] restoreConfig: found saved original Bazel path ${saved.bazelPath}")
+            configState.blazeBinaryState.blazeBinary = saved.bazelPath
+        }
+    }
+
+    override fun processTerminating(executorId: String, env: ExecutionEnvironment, handler: ProcessHandler) {
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processTerminating: $executorId $env $handler")
+
+        val state = getBlazeConfigurationState(env) ?: run {
+            MirrordLogger.logger.debug("[${this.javaClass.name}] processTerminating: Bazel not detected")
+            return
+        }
+
+        restoreConfig(executorId, state)
+
+        super.processTerminating(executorId, env, handler)
+    }
+
+    override fun processNotStarted(executorId: String, env: ExecutionEnvironment) {
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processStarted (noop): $executorId $env")
+        super.processNotStarted(executorId, env)
+    }
+
+    override fun processStarted(executorId: String, env: ExecutionEnvironment, handler: ProcessHandler) {
+        MirrordLogger.logger.debug("[${this.javaClass.name}] processStarted (noop): $executorId $env $handler")
+        super.processStarted(executorId, env, handler)
+    }
```

Comment:
why override those methods?

Reply:
Dunno, I guess just for the extra logs. I copied code of the tomcat listener and worked upon that. Do you want to remove these? I don't think we need them